src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,159 @@
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (isIndexOutForOtherMethods(index)) {
+            throw new IndexOutOfBoundsException();
+        }
+
+        Node<T> current = getNodeForSetAndGetMethods(index);
+
+        return current.value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        if (isIndexOutForOtherMethods(index)) {
+            throw new IndexOutOfBoundsException();
+        }
+
+        Node<T> current = getNodeForSetAndGetMethods(index);
+
+        T oldValue = current.value;
+        current.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (isIndexOutForOtherMethods(index)) {
+            throw new IndexOutOfBoundsException();
+        }
+
+        if (index == 0) {
+            T removedValue = head.value;
+            head = head.next;
+            size--;
+            return removedValue;
+        } else {
+            Node<T> current = getNode(index);
+
+            if (current != null && current.next != null) {
+                T removedValue = current.next.value;
+                current.next = current.next.next;
+                size--;
+                return removedValue;
+
+            } else {
+                throw new NoSuchElementException("Element not found");
+            }
+        }
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> current = head;
+        Node<T> prev = null;
+
+        while (current != null) {
+            if ((object == null && current.value == null)
+                    || (object != null && object.equals(current.value))) {
+                if (prev == null) {
+                    head = current.next;
+                } else {
+                    prev.next = current.next;
+                }
+                size--;
+                return true;
+            }
+            prev = current;
+            current = current.next;
+        }
+
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+
+    }
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> next;
+        private Node<T> prev;
+
+        Node(Node<T> prev, T value, Node<T> next) {
+            this.next = next;
+            this.value = value;
+            this.prev = prev;
+        }
+    }
+
+    private boolean isIndexOutForAddMethod(int index) {
+        return index < 0 || index > size;
+    }
+
+    private boolean isIndexOutForOtherMethods(int index) {
+        return index < 0 || index >= size;
+    }
+
+    private Node<T> getNodeForSetAndGetMethods(int index) {
+        Node<T> current = head;
+        for (int i = 0; i < index; i++) {
+            current = current.next;
+        }
```

Comment:
you need only one getNode method. Valid index condition is differenf for add in collections, but that doesn't mean we need to have separate getnode methods